pkg/kubelet/dockertools/docker.go

```diff
@@ -1049,3 +1049,21 @@ func RunContainer(client DockerInterface, container *api.Container, pod *api.Pod
 	}
 	return dockerContainer.ID, nil
 }
+
+// KillContainer kills a docker container with the given containerID.
+// TODO(yifan): To use strong type for the containerID.
+func KillContainer(client DockerInterface, containerID string,
```

Comment:
@vmarmol Thank you. I am thinking that maybe we don't need to let container runtime manage the restart of the whold pod. Maybe we can figure out a way that let the containers restart during the reconciliation process.
For example:

```
docker.RunContainer() {
    if needRestartPodInfraContainer {
        restartPodInfraContainer()
        invalidateRunningPod().
    }
}

kubelet.syncPod()
    for each container in runningPod:
        restart := hashChanged || unhealthy || invalid
        if restart {
            restartContainer()
        }
    }       
}
```

By this we can pull the ref, readiness, etc out :)